tvl2018.py

```diff
@@ -0,0 +1,1269 @@
+                                 "numpy array, or one of the provided synthesized "
+                                 "sounds (e.g., 'synthesize_{}khz_{}ms').")
+        else:
+            # Assume it's a path to audio file and not a request for synthesized sound
+            if not os.path.exists(filename_or_sound):
+                raise FileNotFoundError("No file found. filename_or_sound must be a string, "
+                                        "numpy array, or one of the provided synthesized sounds "
+                                        "(e.g., 'synthesize_{}khz_{}ms').")
+            data, rate = read_and_resample(filename_or_sound)
+    elif isinstance(filename_or_sound, np.ndarray):
+        file_name = "User Input Audio Data Array"
+        if not rate:
+            raise ValueError("Rate must be specified when providing sound data directly.")
+        data = filename_or_sound
+
+    # Filter the sound field to cochlea
+    data = sound_field_to_cochlea(data, filename_filter)
+
+    # Calculate instantaneous specific loudness
+    instantaneous_specific_loudness_left, instantaneous_specific_loudness_right = \
+        filtered_signal_to_monaural_instantaneous_specific_loudness(data, rate, db_max)
+
+    # Calculate short-term specific loudness for left and right
+    short_term_specific_loudness_left,\
+        short_term_loudness_left = \
+        instantaneous_specific_loudness_to_shortterm_specific_loudness(
+            instantaneous_specific_loudness_left)
+    short_term_specific_loudness_right,\
+        short_term_loudness_right = \
+        instantaneous_specific_loudness_to_shortterm_specific_loudness(
+            instantaneous_specific_loudness_right)
+
+    for i in range(short_term_specific_loudness_left.shape[0]):
+        _, short_term_loudness_left[i], short_term_loudness_right[i] = \
+          monaural_specific_loudness_to_binaural_loudness_025(
+          short_term_specific_loudness_left[i, :],
+          short_term_specific_loudness_right[i, :]
+        )
+
+    long_term_loudness_left = shortterm_loudness_to_longterm_loudness(
+        short_term_loudness_left.flatten())
+    long_term_loudness_right = shortterm_loudness_to_longterm_loudness(
+        short_term_loudness_right.flatten())
+
+    long_term_loudness = long_term_loudness_left + long_term_loudness_right
+    loudness = np.max(long_term_loudness)
+
+    short_term_loudness = \
+        short_term_loudness_left.flatten() + short_term_loudness_right.flatten()
+
+    # Plotting the results
+    plt.figure()
+    plt.plot(range(len(short_term_loudness)), short_term_loudness,
+             'b-', label='Short-term loudness')
+    plt.plot(range(len(long_term_loudness)), long_term_loudness,
+             'r-', label='Long-term loudness')
+    plt.xlabel('time [ms]')
+    plt.ylabel('Loudness [sone]')
+    plt.legend()
+    plt.show()
+
+    # Writing results to text file
```

Comment:
I'd put the rest of this function in a if output_path block.. and tell people that they can use '.' as the directory if they really want the current directory.  Then it is optional.  Writing the results into a text file is very un-pythonic.  :-)